pkg/controller/deployment/deployment_controller.go

```diff
@@ -416,6 +416,23 @@ func (dc *DeploymentController) handleOverlap(d *extensions.Deployment) error {
 	return nil
 }
 
+// overlapsWith returns true when two given deployments are different and overlap with each other
+func overlapsWith(current, other *extensions.Deployment) (bool, error) {
+	if current.UID == other.UID {
+		return false, nil
+	}
+	currentSelector, err := unversioned.LabelSelectorAsSelector(current.Spec.Selector)
+	if err != nil {
+		return false, fmt.Errorf("deployment %s/%s has invalid label selector: %v", current.Namespace, current.Name, err)
+	}
+	otherSelector, err := unversioned.LabelSelectorAsSelector(other.Spec.Selector)
+	if err != nil {
+		return false, fmt.Errorf("deployment %s/%s has invalid label selector: %v", other.Namespace, other.Name, err)
```

Comment:
This means that broken selectors from other deployments will block the current deployment from being reconciled. Log this and return false.